scripts/core/editor3/components/handlePastedText.ts

```diff
@@ -42,34 +65,36 @@ export function handlePastedText(text, _html) {
     const {editorState, suggestingMode, onPasteFromSuggestingMode} = this.props;
 
     if (!html && !text) {
-        return HANDLED;
+        return DraftHandleValue.HANDLED;
     }
 
     if (suggestingMode) {
         if (!Suggestions.allowEditSuggestionOnLeft(editorState, author)
             && !Suggestions.allowEditSuggestionOnRight(editorState, author)) {
-            return HANDLED;
+            return DraftHandleValue.HANDLED;
         }
 
         const content = html ? getContentStateFromHtml(html) : ContentState.createFromText(text);
 
         onPasteFromSuggestingMode(content);
-        return HANDLED;
+        return DraftHandleValue.HANDLED;
     }
 
+    if (pasteContentFromOpenEditor(this.props, html) === DraftHandleValue.HANDLED) {
+        return DraftHandleValue.HANDLED;
+    }
+
+
     if (htmlComesFromDraftjsEditor(html)) {
-        return NOT_HANDLED;
+        return DraftHandleValue.NOT_HANDLED;
     }
 
     return processPastedHtml(this.props, html || text);
 }
 
-// Checks if there are atomic blocks in the paste content. If there are, we need to set
-// the 'atomic' block type using the Modifier tool and add these entities to the
-// contentState.
-function processPastedHtml(props, html) {
-    const {onChange, editorState, editorFormat} = props;
-    let pastedContent = getContentStateFromHtml(html);
+function insertContentInState(props: any, _pastedContent: ContentState) : DraftHandleValue {
```

Comment:
Instead of props, take the actual arguments you need: `editorState, editorFormat, onChange`. This way function signature becomes cleaner and you can add type annotations without needing to add it for the entire object.

It also makes it easier to understand and refactor the code if you take fewer arguments. For example `pasteContentFromOpenEditor` which calls this function(`insertContentInState`) takes props as an argument, and while I'm looking at `pasteContentFromOpenEditor` I'm thinking "why does it take all props as an argument? And then I see: oh, it passes the entire object to another function, it must require it then I think. In this case, it's only two levels, but when you keep doing it, you end up with 6 levels of passing the entire object when you only need a few params from it which obfuscates your intention and creates dependencies(on the entire object) which should not be there.

Then comes a time when you need to call that function and you have those 3 required values(`editorState, editorFormat, onChange`), but not the entire object and you think: "I'll just create an empty object with these 3 values and it will work for that function". And then someone modifying the function thinks "I've got the entire object, I can start using another property" - and the errors start coming.

That's probably more context than required, but I saw these things happen in the code and it's quite simple to avoid it.

By the way, if you use `props` outside a React component, specify what kind props is that. It's obvious while in a React component, but otherwise it looks like `_this` as an argument which doesn't really mean anything outside of the class.

Reply:
Yeah you're right. I didn't think about it as I didn't write `insertContentInState` (just renamed it) but I'm working on it right now while I fix other types. Thanks